bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/DbLedgerStorage.java

```diff
@@ -712,23 +874,38 @@ public EntryLogger getEntryLogger() {
 
     @Override
     public long getLastAddConfirmed(long ledgerId) throws IOException {
-        throw new UnsupportedOperationException();
+        TransientLedgerInfo ledgerInfo = transientLedgerInfoCache.getIfPresent(ledgerId);
+        Long lac = null != ledgerInfo ? ledgerInfo.getLastAddConfirmed() : null;
+        if (null == lac) {
+            ByteBuf bb = getEntry(ledgerId, BookieProtocol.LAST_ADD_CONFIRMED);
```

Comment:
It'd be better to do this in the loader, and not rely on a getLastAddConfirmed coming before getExplicitLAC or any of the other calls. If it's in the loader, it becomes easier to reason about the state. This method becomes 
```
return transientLedgerInfoCache.getUnchecked(ledgerId).getLastAddConfirmed();
```
No checking, because we can assume ledger info cache is up to date.

Reply:
no. I have kept the cache as simple and the loading will never throw any exceptions. If I move the getEntry as part of loader, it complicates error handling in cache loading, which I don't want to.

Reply:
fair enough.